contracts/fee/FeeHandler.sol

```diff
@@ -3,46 +3,176 @@
+    }
+
+    function _setAvailableFeeAmount(address feeToken, address buybackToken, uint256 availableFeeAmount) private {
+        dataStore.setUint(Keys.buybackAvailableFeeAmountKey(feeToken, buybackToken), availableFeeAmount);
+    }
+
+    function _getAvailableFeeAmount(address feeToken, address buybackToken) private view returns (uint256) {
+        uint256 availableFeeAmount = dataStore.getUint(Keys.buybackAvailableFeeAmountKey(feeToken, buybackToken));
+        return availableFeeAmount;
+    }
+
+    function _getBuybackGmxFactor(uint256 version) private view returns (uint256) {
+        uint256 buybackGmxFactor = dataStore.getUint(Keys.buybackGmxFactorKey(version));
+        return buybackGmxFactor;
+    }
+
+    function _getMinFeeTokenAmount(
+        address feeToken,
+        address buybackToken,
+        uint256 batchSize
+    ) private view returns (uint256) {
+        (
+            uint256 feeTokenOraclePrice,
+            uint256 feeTokenOracleDecimals,
+            uint256 feeTokenDecimals
+        ) = _getTokenPricingValues(feeToken);
+
+        (
+            uint256 buybackTokenOraclePrice,
+            uint256 buybackTokenOracleDecimals,
+            uint256 buybackTokenDecimals
+        ) = _getTokenPricingValues(buybackToken);
+
+        if (feeTokenOracleDecimals > buybackTokenOracleDecimals) {
+            buybackTokenOraclePrice =
+                buybackTokenOraclePrice *
+                (10 ** (feeTokenOracleDecimals - buybackTokenOracleDecimals));
+        } else if (buybackTokenOracleDecimals > feeTokenOracleDecimals) {
+            feeTokenOraclePrice = feeTokenOraclePrice * (10 ** (buybackTokenOracleDecimals - feeTokenOracleDecimals));
+        }
+
+        uint256 feeTokenPriceInBuybackToken = (feeTokenOraclePrice * 1e30) / buybackTokenOraclePrice;
+
+        // this assumes the batchSize amount is stored in buybackTokenDecimals i.e. batchSize of 100 GMX stored as 100 * (10 ^ 18)
+        if (buybackTokenDecimals > feeTokenDecimals) {
+            batchSize = batchSize / (10 ** (buybackTokenDecimals - feeTokenDecimals));
+        } else if (feeTokenDecimals > buybackTokenDecimals) {
+            batchSize = batchSize * (10 ** (feeTokenDecimals - buybackTokenDecimals));
         }
+
+        uint256 feeTokenAmount = (batchSize * feeTokenPriceInBuybackToken) / 1e30;
+        return feeTokenAmount;
+    }
+
+    function _getTokenPricingValues(address token) private view returns (uint256, uint256, uint256) {
+        address tokenPriceFeedAddress = dataStore.getAddress(Keys.priceFeedKey(token));
+        uint256 tokenOraclePrice = IFeedAddress(tokenPriceFeedAddress).latestAnswer();
+        uint256 tokenOracleDecimals = IFeedAddress(tokenPriceFeedAddress).decimals();
+        uint256 tokenDecimals = IFeedAddress(token).decimals(); // IERC20().decimals() is not included in the openzeppelin IERC20 interface so using IFeedAddress().decimals() instead but can change if necessary
```

Comment:
there should not be a need to read token decimals if the prices are already adjusted: https://github.com/gmx-io/gmx-synthetics/blob/main/contracts/oracle/ChainlinkPriceFeedUtils.sol#L62

Reply:
As per the above comment, this is no longer needed and has been removed.